Seagate/cortx-prvsnr
# [testing] Test automation setup provisioner

- adds automated test for setup_provisioner
- the tests are parametrized per:
  - run mode: remote api / remote cli / on target host cli
  - source type: local / iso
  - number of nodes: 1 / 3 / 6
  - ha / no-ha
- how to try / test:
  - to check available tests `pytest test/api/python/integration/setup/test_setup.py --collect-only`
  - to run all `pytest test/api/python/integration/setup/test_setup.py`
    - **Note**. some of them would fail since not all are workable yet
    - **Note**. depending on your HW the whole set of tests (about 30) would require much time (few hours) 
  - to run specific one, e.g. `pytest test/api/python/integration/setup/test_setup.py --log-cli-level INFO -k test_setup_provisioner_api[SourceT.LOCAL-noHA-ScaleFactorT.NODE1]`


-----
[View rendered docs/testing.md](https://github.com/andkononykhin2/cortx-prvsnr/blob/test-automation-setup-provisioner/docs/testing.md)
